### PR TITLE
Add testable degen gambit file

### PR DESCRIPTION
**Pull request by mrk-hub**
Created TestableDegenGambit.sol
Purpose to allow for easier testing off and on-testchains.
Function List:
setEntropy,
_entropy,
mint,
setDailyStreak,
setWeeklyStreak,
generateEntropyForUnmodifiedReelOutcome,
generateEntropyForImprovedReelOutcome,
setEntropyFromOutcomes

### CODE REVIEW

File: bindings/Tests/TestableDegenGambit/TestableDegenGambit.go
Comment (zomglings) [inline bindings/Tests/TestableDegenGambit/TestableDegenGambit.go:1]: I would do `bindings/TestableDegenGambit/TestableDegenGambit.go`.

Current structure is a bit misleading - when I see `Tests`, I think there will be actual tests in there, which there aren't.

File: Makefile [hunk 1]
```diff
@@ -14,6 +14,10 @@ bindings/BlockInspector/BlockInspector.go:
 	mkdir -p bindings/BlockInspector
 	seer evm generate --package BlockInspector --output bindings/BlockInspector/BlockInspector.go --foundry out/BlockInspector.sol/BlockInspector.json --cli --struct BlockInspector
 
+bindings/Tests.go:
```
Comment (zomglings) [inline Makefile:17]: This should be the path to the output file. `make` will check if the file at this path exists and only run this target if it doesn't.

If you want it to be a target which always runs, you need to list it under .PHONY at the top of the file.

Here's some documentation on `make`: https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

File: cmd/test/cmd.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/PermissionlessGames/degen-casino/bindings/Tests/TestableDegenGambit"
+	"github.com/PermissionlessGames/degen-casino/version"
+)
+
+func CreateRootCommand() *cobra.Command {
+	// rootCmd represents the base command when called without any subcommands
+	rootCmd := &cobra.Command{
+		Use:   "test",
```
Comment (zomglings) [inline cmd/test/cmd.go:15]: `test` is so misleading. Can you add this in `technician`?

File: src/DegenGambit.sol [hunk 3]
```diff
@@ -795,4 +795,15 @@ contract DegenGambit is ERC20, ReentrancyGuard {
 
         prize = payout(left, center, right);
     }
+
+    /// isDebuggable pure function that returns a string with debug version
+    /// if empty string non-debuggable
+    function debuggerVersion()
```
Comment (zomglings) [inline src/DegenGambit.sol:801]: I don't like that the main contract has something called `debuggerVersion`.

Why not have a simple `version()` method?

For production contracts, we could just have versions like `"1"`, `"42"`, etc. (I don't think we need semantic versioning here.)

For debuggable contracts, we can use `"1-debug"`, `"42-debug"`, etc.

Clients can parse these to decide on behavior.

File: src/testable/TestableDegenGambit.sol [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    function setEntropy(address player, uint256 entropy) public {
+        EntropyForPlayer[player] = entropy;
+    }
+
+    function _entropy(address player) internal view override returns (uint256) {
+        return EntropyForPlayer[player];
+    }
+
+    function mint(address to, uint256 amount) public {
```
Comment (zomglings) [inline src/testable/TestableDegenGambit.sol:23]: Maybe let's call this `mintGambit`?

File: src/testable/TestableDegenGambit.sol [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            : generateEntropyForUnmodifiedReelOutcome(left, center, right);
+        EntropyForPlayer[player] = entropy;
+    }
+
+    function generateEntropyForImprovedReelOutcome(
+        uint256 leftOutcome,
+        uint256 centerOutcome,
+        uint256 rightOutcome
+    ) public view returns (uint256) {
+        // Ensure the outcome indices are within the valid range (0-17)
```
Comment (zomglings) [inline src/testable/TestableDegenGambit.sol:98]: Why is the range different here? Shouldn't it be 0 - 18?

**Review by zomglings — CHANGES_REQUESTED** [review]
This is missing tests for `TestableDegenGambit` - please add tests that when you set the appropriate things the correct state is affected on the contract.

Several other suggestions inline.

File: cmd/test/cmd.go (hunk 2 above)
Comment (mrk-hub) [reply cmd/test/cmd.go:15]: Yes, I can move this into technician instead of having a seperate cmd system

File: Makefile (hunk 1 above)
Comment (mrk-hub) [reply Makefile:17]: I renamed the path and remove test from bin

File: src/DegenGambit.sol (hunk 3 above)
Comment (mrk-hub) [reply src/DegenGambit.sol:801]: Took the suggestion made it version instead. Looks cleaner and nicer

File: src/testable/TestableDegenGambit.sol (hunk 5 above)
Comment (mrk-hub) [reply src/testable/TestableDegenGambit.sol:98]: Fixed.

**Pull request merged**